### PR TITLE
Remove .jl extension

### PR DESCRIPTION
**Pull request by fredrikekre**
ref https://github.com/JuliaLang/julia/issues/20630#issuecomment-280423789

~~requires https://github.com/JuliaLang/julia/pull/20681~~

### CODE REVIEW

**Comment by tkelman** [comment]
would prefer to maintain 0.5 support here

**Comment by yuyichao** [comment]
Seems like a trivial function to just include here instead of using a internal function in Base.

**Comment by fredrikekre** [comment]
Yea, that's a good idea.

File: test/runtests.jl [hunk 1]
```diff
@@ -174,10 +178,10 @@ end"""
         Pkg.add("Example")
         io = IOBuffer()
         f = PkgDev.freeable(io)
-        @test !(any(f .== "Example") || contains(takebuf_string(io), "Example"))
+        @test !(any(f .== "Example") || contains(String(take!(io)), "Example"))
```
Comment (tkelman) [inline test/runtests.jl]: this'll need Compat min version to be raised to 0.9.5

Comment (fredrikekre) [inline test/runtests.jl]: Thanks.

File: test/runtests.jl [hunk 2]
```diff
@@ -52,7 +52,11 @@ temp_pkg_dir() do pkgdir
         end
     end
 
-
+    @testset "generating a package with .jl extension" begin
+        PkgDev.generate("PackageWithExtension.jl", "MIT", config=Dict("user.name"=>"Julia Test", "user.email"=>"test@julialang.org"))
+        @show keys(Pkg.installed())
+        @test in("PackageWithExtension", [keys(Pkg.installed())...])
```
Comment (ararslan) [inline test/runtests.jl]: You shouldn't need to collect the key iterator into an array; `@test "PackageWithExtension" in keys(Pkg.installed())` should be fine.

Comment (fredrikekre) [inline test/runtests.jl]: I copied from the test above, but I realize now that the other test collects to make sure thats the only installed package. Will update!

**Comment by fredrikekre** [comment]
Travis nightlies:
```
testing freeable: Error During Test
  Got an exception of type MethodError outside of a @test
  MethodError: no method matching get(::Nullable{Base.Pkg.Types.Available})
  You may have intended to import Base.get
  Closest candidates are:
    get{T<:AbstractString}(::Type{T<:AbstractString}, ::Base.LibGit2.GitConfig, ::AbstractString) at libgit2/config.jl:55
    get(::Type{Bool}, ::Base.LibGit2.GitConfig, ::AbstractString) at libgit2/config.jl:65
    get(::Type{Int32}, ::Base.LibGit2.GitConfig, ::AbstractString) at libgit2/config.jl:72
    ...
  Stacktrace:
   [1] (::PkgDev.Entry.##53#56{Base.AbstractIOBuffer{Array{UInt8,1}},PkgDev.Entry.#latest_tag#54,Array{Any,1}})(::Base.LibGit2.GitRepo) at /home/travis/.julia/v0.6/PkgDev/src/entry.jl:322
   [2] with(::PkgDev.Entry.##53#56{Base.AbstractIOBuffer{Array{UInt8,1}},PkgDev.Entry.#latest_tag#54,Array{Any,1}}, ::Base.LibGit2.GitRepo) at ./libgit2/types.jl:576
   [3] (::PkgDev.Entry.##52#55{Base.AbstractIOBuffer{Array{UInt8,1}},PkgDev.Entry.#latest_tag#54,Array{Any,1}})() at /home/travis/.julia/v0.6/PkgDev/src/entry.jl:318
   [4] cd(::PkgDev.Entry.##52#55{Base.AbstractIOBuffer{Array{UInt8,1}},PkgDev.Entry.#latest_tag#54,Array{Any,1}}, ::String) at ./file.jl:69
   [5] freeable(::Base.AbstractIOBuffer{Array{UInt8,1}}) at /home/travis/.julia/v0.6/PkgDev/src/entry.jl:314
   [6] (::Base.Pkg.Dir.##2#3{Array{Any,1},PkgDev.Entry.#freeable,Tuple{Base.AbstractIOBuffer{Array{UInt8,1}}}})() at ./pkg/dir.jl:31
   [7] cd(::Base.Pkg.Dir.##2#3{Array{Any,1},PkgDev.Entry.#freeable,Tuple{Base.AbstractIOBuffer{Array{UInt8,1}}}}, ::String) at ./file.jl:69
   [8] #cd#1(::Array{Any,1}, ::Function, ::Function, ::Base.AbstractIOBuffer{Array{UInt8,1}}, ::Vararg{Base.AbstractIOBuffer{Array{UInt8,1}},N} where N) at ./pkg/dir.jl:31
   [9] macro expansion at /home/travis/.julia/v0.6/PkgDev/test/runtests.jl:180 [inlined]
   [10] macro expansion at ./test.jl:852 [inlined]
   [11] (::##1#11)(::String) at /home/travis/.julia/v0.6/PkgDev/test/runtests.jl:177
   [12] temp_pkg_dir(::##1#11, ::Bool) at /home/travis/.julia/v0.6/PkgDev/test/runtests.jl:15
   [13] temp_pkg_dir(::Function) at /home/travis/.julia/v0.6/PkgDev/test/runtests.jl:8
   [14] include_from_node1(::String) at ./loading.jl:539
   [15] include(::String) at ./sysimg.jl:14
   [16] process_options(::Base.JLOptions) at ./client.jl:300
   [17] _start() at ./client.jl:366
Test Summary:    | Error  Total
testing freeable |     1      1
ERROR: LoadError: Some tests did not pass: 0 passed, 0 failed, 1 errored, 0 broken.
while loading /home/travis/.julia/v0.6/PkgDev/test/runtests.jl, in expression starting on line 21
```

AV timed out.

**Comment by fredrikekre** [comment]
Could just restart, doesn't take long.

**Comment by ararslan** [comment]
I can't restart AppVeyor since that runs on Art's personal account, so I'll close and reopen to retrigger all CI.

**Comment by ararslan** [comment]
Looks like this is passing on nightly (at least as of this writing it's passed on Linux nightly), so unless there are any objections I'll merge once CI is green.

**Comment by tkelman** [comment]
appveyor isn't running on release, should be sure that's the case

**Comment by ararslan** [comment]
Should be sure what's the case?

Also it seems 32-bit AppVeyor is likely to time out; it's doing the coverage freakout thing that it seems to do on occasion.

**Comment by tkelman** [comment]
that appveyor is running against release julia

**Comment by ararslan** [comment]
AV is only set up to run on nightly: https://github.com/JuliaLang/PkgDev.jl/blob/master/appveyor.yml

So we should add 0.5 to the AV yaml then restart this?

**Comment by fredrikekre** [comment]
This?
```diff
diff --git a/appveyor.yml b/appveyor.yml
index 324b423..d09642a 100644
--- a/appveyor.yml
+++ b/appveyor.yml
@@ -1,10 +1,13 @@
 environment:
   matrix:
+  - JULIAVERSION: "julialang/bin/winnt/x86/0.5/julia-0.5-latest-win32.exe"
+  - JULIAVERSION: "julialang/bin/winnt/x64/0.5/julia-0.5-latest-win64.exe"
   - JULIAVERSION: "julianightlies/bin/winnt/x86/julia-latest-win32.exe"
   - JULIAVERSION: "julianightlies/bin/winnt/x64/julia-latest-win64.exe"
 
 branches:
   only:
+    - /release-.*/
     - master
 
 notifications:
```

**Comment by yuyichao** [comment]
https://github.com/JuliaLang/PkgDev.jl/pull/102.   No need to add the branch

**Comment by fredrikekre** [comment]
Ah, nice.

**Comment by ararslan** [comment]
Julia 0.5 has been added to the AppVeyor setup, so I'll retrigger all CI again.

**Comment by ararslan** [comment]
CI doesn't seem to have been retriggered...?

**Comment by fredrikekre** [comment]
Let's re trigger now that AWS S3 are up.

**Comment by ararslan** [comment]
Nightly AppVeyor failure is #103.

**Pull request merged**